ops/testing.py

```diff
@@ -739,11 +743,12 @@ def attach_storage(self, storage_id: str) -> None:
         """Attach a storage device.
 
         The intent of this function is to simulate a ``juju attach-storage`` call.
-        It will trigger a storage-attached hook if the storage unit in question exists
+        If called after :meth:`begin` and hooks are not disabled, it will trigger
+        a storage-attached hook if the storage unit in question exists
         and is presently marked as detached.
 
         The test harness uses symbolic links to imitate storage mounts, which may lead to some
-        inconsistencies compared to the actual charm. Users should be cognizant of
+        inconsistencies compared to the actual charm. Users should be cognisant of
```

Comment:
Thanks for the spelling fix. However, while we're here...

I am allergic to the phrase "cognisant of" (I had a coworker who'd use it in every other sentence at my previous workplace :-). "aware of" would be fine, however, in this case I think we should just drop this last sentence -- we've just made them aware of it in the previous sentence.

Reply:
:smile: I considered just replacing it with "aware", which is much more understandable. Dropping sounds good to me.